src/ct_eval.cpp

```diff
@@ -170,12 +170,12 @@ TypeFunctionId type_func_from_ast(TypedAST::TypedAST* ast, TypeChecker& tc) {
 	} else if (ast->type() == TypedASTTag::StructExpression) {
 		auto as_se = static_cast<TypedAST::StructExpression*>(ast);
 
-		std::vector<std::string> fields;
-		std::unordered_map<std::string, MonoId> structure;
+		std::vector<InternedString> fields;
+		std::unordered_map<InternedString, MonoId> structure;
 		int field_count = as_se->m_fields.size();
 		for (int i = 0; i < field_count; ++i){
 			MonoId mono = mono_type_from_ast(as_se->m_types[i], tc);
-			std::string name = as_se->m_fields[i].text().str();
+			InternedString name(as_se->m_fields[i].text().str());
```

Comment:
`Identifier::text` returns an InternedString. you don't need to call `.str()` here

```c++
auto name = as_se->m_fields[i].text();
```

should do the same thing (if a little bit faster)